mods/fire/init.lua

```diff
@@ -109,27 +54,46 @@ minetest.register_node("fire:permanent_flame", {
 	sunlight_propagates = true,
 	floodable = true,
 	damage_per_second = 4,
-	groups = {igniter = 2, dig_immediate = 3},
+	groups = {igniter = 2, dig_immediate = 3, fire = 1},
 	drop = "",
+	on_flood = flood_flame
+}
+
+-- Permanent flame node
+local permanent_fire_node = table.copy(fire_node)
+permanent_fire_node.description = S("Permanent Flame")
+permanent_fire_node.groups.not_in_creative_inventory = 1
+
+minetest.register_node("fire:permanent_flame", permanent_fire_node)
+
+-- Basic flame node
+local flame_fire_node = table.copy(fire_node)
+flame_fire_node.on_timer = function(pos)
+	local f = minetest.find_node_near(pos, 1, {"group:flammable"})
+	if not f then
```

Comment:
Why was the check for `fire_enabled` removed here?

Reply:
To not delete become decorative fire after disabling this setting. It will not burn anything.